ledger/src/blockstore_processor.rs

```diff
@@ -1218,15 +1218,10 @@ impl TransactionStatusSender {
         balances: TransactionBalancesSet,
         token_balances: TransactionTokenBalancesSet,
         inner_instructions: Vec<Option<InnerInstructionsList>>,
-        transaction_logs: Vec<TransactionLogMessages>,
+        transaction_logs: Vec<Option<TransactionLogMessages>>,
         rent_debits: Vec<RentDebits>,
     ) {
         let slot = bank.slot();
-        let (inner_instructions, transaction_logs) = if !self.enable_cpi_and_log_storage {
```

Comment:
As context, using the Option to pass into the TransactionStatusService was a contortion to avoid having to collect empty Vecs for when `--enable-cpi-and-log-storage` is not present: https://github.com/solana-labs/solana/pull/14922#discussion_r566706946

Reply:
Thanks for catching this! I thought this flag was redundant with the `enable_cpi_recording` and `enable_log_recording` flags. I added it back and I believe the option handling should match the previous behavior as well.

Reply:
Thanks!